awx/api/views/__init__.py

```diff
@@ -371,13 +371,38 @@ def get_queryset(self):
 
 
 class InstancePeersList(SubListAPIView):
-    name = _("Instance Peers")
+    name = _("Peers")
     parent_model = models.Instance
-    model = models.Instance
-    serializer_class = serializers.InstanceSerializer
+    model = models.ReceptorAddress
+    serializer_class = serializers.ReceptorAddressSerializer
     parent_access = 'read'
-    search_fields = {'hostname'}
     relationship = 'peers'
+    search_fields = 'address'
+
+
+class InstanceReceptorAddressesList(ListCreateAPIView):
+    name = _("Receptor Addresses")
+    model = models.ReceptorAddress
+    serializer_class = serializers.ReceptorAddressSerializer
+
+    def get_queryset(self):
+        return models.ReceptorAddress.objects.filter(instance=self.kwargs['pk'])
+
+    def post(self, request, *args, **kwargs):
+        request.data.update({'instance': self.kwargs['pk']})
+        return super().post(request, *args, **kwargs)
+
+
+class ReceptorAddressesList(ListAPIView):
```

Comment:
Clients have a hard time when a resource can be created in a sublist, but not in a global list. Since address->instance relationship is a ForeignKey, the contract is easy to manage. Just make sure `null=False` is set on the `instance` field, and the on_delete behavior is already good.